db/schema.rb

```diff
@@ -33,6 +33,16 @@
     t.index ["key"], name: "index_active_storage_blobs_on_key", unique: true
   end
 
+  create_table "character_attributes", force: :cascade do |t|
+    t.integer "character_type_id", null: false
+    t.string "name"
+    t.integer "value"
```

Comment:
You should ensure the presence of those attributes on the DB level also. ActiveRecord validations can be skipped if a query is executed on DB directly.